### ISSUE TITLE
[Windows] Generating custom dataset fails

### ISSUE THREAD

**Issue by Brannonj96** [snipsco/snips-nlu#605]
I'm following the tutorial, so I have three intent files, setTemperature.txt, turnLightsOn.txt, and turnLightsOff.txt, and one entity file, rooms.txt.  Running

`generate-dataset --language en --intent-files   turnLightOn.txt turnLightOff.txt setTemperature.txt --entity-files room.txt > dataset.json`

generates a dataset, but when I do 

```
 with io.open("dataset.json") as f:
    dataset = json.load(f)
engine.fit(dataset)

```
I get the following error

```
Traceback (most recent call last):
  File "C:\Users\J90779\.spyder-py3\testSnipIdle.py", line 12, in <module>
    engine.fit(dataset)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\utils.py", line 259, in wrapped
    res = fn(*args, **kwargs)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\nlu_engine\nlu_engine.py", line 95, in fit
    recycled_parser.fit(dataset, force_retrain)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\utils.py", line 259, in wrapped
    res = fn(*args, **kwargs)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\intent_parser\probabilistic_intent_parser.py", line 84, in fit
    self.slot_fillers[intent_name].fit(dataset, intent_name)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\utils.py", line 259, in wrapped
    res = fn(*args, **kwargs)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\slot_filler\crf_slot_filler.py", line 133, in fit
    for sample in crf_samples]
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\slot_filler\crf_slot_filler.py", line 133, in <listcomp>
    for sample in crf_samples]
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\slot_filler\crf_slot_filler.py", line 203, in compute_features
    value = feature.compute(i, cache)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\slot_filler\feature.py", line 59, in compute
    value = self.function(tokens, token_index + self.offset)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\slot_filler\feature_factory.py", line 498, in builtin_entity_match
    text, self.language, scope=[builtin_entity], use_cache=True)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\builtin_entities.py", line 46, in get_builtin_entities
    return parser.parse(text, scope=scope, use_cache=use_cache)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu\builtin_entities.py", line 26, in parse
    parser_result = self.parser.parse(text, scope)
  File "C:\Users\J90779\AppData\Local\Programs\Python\Python36\lib\site-packages\snips_nlu_ontology\builtin_entities.py", line 159, in parse
    self._parser, text.encode("utf8"), scope, byref(ptr))
OSError: [WinError -529697949] Windows Error 0xe06d7363
```

Everything is in the same folder, and fitting the sample_dataset.json works just fine.  I also tried to manually type out a dataset following the formatting in sample_dataset, but I get the same error.  Is this an issue on my side or something else. 

Also running snips version 0.14.0

**Comment by adrienball** [comment]
Hey @Brannonj96, I didn't have time to setup a Windows VM on my machine in order to debug windows specific issues, that's on my todo list.
In the meantime, can you try to replicate the issue with a tiny dataset that you could share in this thread (through drag & drop) ? That would allow me to double check that it is indeed a windows specific issue and not something else.
Thanks!

**Comment by Brannonj96** [comment]
Here is a JSON that gives the error in .txt form (github doesn't allow .json uploaded and compressing one small file to a .zip seemed unnecessary).

[testSet.txt](https://github.com/snipsco/snips-nlu/files/2125341/testSet.txt)


I have narrowed it down to the snips/temperature builtin entity.  If I remove every slot that uses it, the generation works fine.  I haven't been able to create a proper use of it without it giving the same error.  I've tested it where it was just "19" not "19 degrees" tested it with various layouts, but calling snips/temperature gives that windows os error.  

It may be some weird interaction when it actually tries to use the entity.  Other built in entities seem to work (at least datetime does).  Auto-generated and hand-made .json files fail with snips/temperature.

**Comment by adrienball** [comment]
Thanks @Brannonj96 ,
I cannot reproduce on my machine (macOS), so I guess it's windows specific. I'll get back to you once I've managed to replicate it on a windows environment.
Cheers

**Comment by steve-bakos** [comment]
I have the same issue. A "work around" I found for it was if you use snips/amountOfMoney as the entity instead of snips/temperature or snips/number, and put a $ in front of the temperature, you can get passed the error and it works as expected.

**Comment by Brannonj96** [comment]
Hey, I figured it out.  If the value in the entity starts with a number it gives this error.

`"test":{
		"automatically_extensible": true,
		"data": [
			{
				"value": "1234 street street",
				"synonyms":[]
			}
		],
		"use_synonyms":false
	}`

will error

`"test":{
		"automatically_extensible": true,
		"data": [
			{
				"value": "street street 1234",
				"synonyms":[]
			}
		],
		"use_synonyms":false
	}`

will not error.  This also corresponds to my temperature error and the aforementioned money error.

**Comment by adrienball** [comment]
Hi @Brannonj96 @steve-bakos, 
I managed to reproduce as well on Windows. From first investigations, it seems to come from some datetime overflows on 32-bits architectures. 
Regarding this, it would be helpful to know which version of Windows you are using, 32 bits or 64 bits.

We made a fix internally, that solved the issue on our windows environment and hopefully on yours as well. I'll keep you posted here about the updates.
Thanks!

**Comment by steve-bakos** [comment]
Hey @adrienball, I'm having this issue on Windows 10 64-bit.

**Comment by Brannonj96** [comment]
And I am using Windows 7 64-bit

**Comment by jeremy-power** [comment]
Hey I just wanted to add that I am also running into this isue on Windows 10 64-bit and was wondering if there were any updates.
Thanks!

**Comment by adrienball** [comment]
Hey @jeremy-power ,
The fix will be shipped in the next release, which I expect to be end of this week or beginning of next week.
Cheers

**Comment by adrienball** [comment]
Reopening as it was closed automatically by github, but not released yet.

**Comment by adrienball** [comment]
Hey @Brannonj96 @steve-bakos @jeremy-power ,
I just released `snips-nlu==0.15.1`. Can you upgrade to the latest version, and check if you still have the crash ?
Thanks!

*Side note:* there are some precompiled wheels on pypi for the target win_amd64, for snips-nlu dependencies, so the package installation should be much faster.

**Comment by Brannonj96** [comment]
Just updated (only took 5-10 seconds), and all the previously failing test sets are passing now.  Thanks for the update!
